### PR TITLE
Add Local Configuration to Development Environment page

### LINKED ISSUE #2521: Support for dotenv
I was going to ask this in Discord but decided to do it here in case others have the same question.

[dotenv](https://www.npmjs.com/package/dotenv) is a popular mechanism of specifying project-specific environment variables without accidentally adding secrets to git history. It relies on git ignored `.env` file with environment variables. 

Backstage has it's own configuration reading API. In theory, dotenv and Backstage config should not be in conflict as long as dotenv is loaded before Backstage config. At the same time, adding secret environment variables directly to `app-config.development.yaml` and git ignoring `app-config.development.yaml` would serve the same functionality as dotenv.

What are your thoughts on how app-config.development.yaml should be used and the role of .env if any?

### PR DESCRIPTION
**Pull request by taras**
Closes #2521 

## Hey, I just made a Pull Request!

I added instructions about `app-config.development.local.yaml` file to the Development Environment section of the site to help people getting started configure their local environment settings.

I added the following text,

```md
## Local configuration

Backstage allows you to specify the configuration used while running the
application on your computer. Local configuration is read from
`app-config.local.yaml`. This file is ignored by Git, which means that you can
safely use it to reference secrets like GitHub tokens without worrying about
these secrets, inadvertently ending up in the Git repository. You do not need to
copy everything from the default config to the local config.
`app-config.local.yaml` will be merged with `app-config.yaml` and overwrite the
default app configs.

You can learn more about the local configuration in
[Static Configuration in Backstage](../conf/) section.
```

#### :heavy_check_mark: Checklist

- [ ] All tests are passing `yarn test`
- [ ] Screenshots attached (for UI changes)
- [ ] Relevant documentation updated
- [ ] Prettier run on changed files
- [ ] Tests added for new functionality
- [ ] Regression tests added for bug fixes

### CODE REVIEW

File: docs/getting-started/development-environment.md [hunk 1]
```diff
@@ -88,3 +88,14 @@ yarn create-plugin # Create a new plugin
...
+
+## Local configuration
+
+Backstage allows you to specify the configuration used while running the
+application on your computer. Local configuration is read from
+`app-config.development.local.yaml`. This file is ignored by Git, which means
+that you can safely use it to reference secrets like GitHub tokens worrying
+about these secrets inadvertently ending up in the Git repository.
+
+You can learn more about local configuration in
```
Comment (freben) [inline docs/getting-started/development-environment.md:100]: +the ("in the")

File: docs/getting-started/development-environment.md [hunk 2]
```diff
@@ -88,3 +88,14 @@ yarn create-plugin # Create a new plugin
 > See
 > [package.json](https://github.com/spotify/backstage/blob/master/package.json)
 > for other yarn commands/options.
+
+## Local configuration
+
+Backstage allows you to specify the configuration used while running the
+application on your computer. Local configuration is read from
+`app-config.development.local.yaml`. This file is ignored by Git, which means
+that you can safely use it to reference secrets like GitHub tokens worrying
```
Comment (freben) [inline docs/getting-started/development-environment.md:97]: +without worrying

File: docs/getting-started/development-environment.md [hunk 3]
```diff
@@ -88,3 +88,14 @@ yarn create-plugin # Create a new plugin
 > See
 > [package.json](https://github.com/spotify/backstage/blob/master/package.json)
 > for other yarn commands/options.
+
+## Local configuration
+
+Backstage allows you to specify the configuration used while running the
+application on your computer. Local configuration is read from
```
Comment (freben) [inline docs/getting-started/development-environment.md:95]: Maybe mention that it merges with the others? So people don't copy over all of it

File: docs/getting-started/development-environment.md [hunk 4]
```diff
@@ -88,3 +88,14 @@ yarn create-plugin # Create a new plugin
...
+`app-config.development.local.yaml`. This file is ignored by Git, which means
```
[the remark this answers is not in the document]
Comment (taras) [reply docs/getting-started/development-environment.md:96]: I'll move it. I wasn't sure.

[the remark this answers is not in the document]
Comment (taras) [reply docs/getting-started/development-environment.md]: Removed in https://github.com/spotify/backstage/pull/2710/commits/65a06ce1e8c5f7c807bc8166b9541850c58697c0

File: docs/getting-started/development-environment.md (hunk 2 above)
Comment (taras) [reply docs/getting-started/development-environment.md]: Added in https://github.com/spotify/backstage/pull/2710/commits/65a06ce1e8c5f7c807bc8166b9541850c58697c0

File: docs/getting-started/development-environment.md (hunk 3 above)
Comment (taras) [reply docs/getting-started/development-environment.md:95]: Added a message about override in https://github.com/spotify/backstage/pull/2710/commits/65a06ce1e8c5f7c807bc8166b9541850c58697c0

**Review by Rugvip — APPROVED** [review]
Nice, thanks!

**Pull request merged**